src/github.com/getlantern/autoupdate-server/server/github_test.go

```diff
@@ -222,7 +240,109 @@ func TestDownloadOldestVersionAndUpgradeIt(t *testing.T) {
 				}
 			}
 
-			// fmt.Printf("r: %v, err: %v\n", r, err)
+			if r.PatchType != PATCHTYPE_BSDIFF {
+				t.Fatal("Expecting no patch.")
+			}
+
+			if r.Version != testClient.latestAssetsMap[os][arch].v.String() {
+				t.Fatal("Expecting %v, got %v.", testClient.latestAssetsMap[os][arch].v, r.Version)
+			}
+		}
+	}
+
+	// Let's walk again using an odd checksum.
+	for os := range oldestVersionMap {
+		for arch := range oldestVersionMap[os] {
+			asset := oldestVersionMap[os][arch]
+			params := Params{
+				AppVersion: asset.v.String(),
+				OS:         asset.OS,
+				Arch:       asset.Arch,
+				Checksum:   "?",
+			}
+
+			r, err := testClient.CheckForUpdate(&params)
+			if err != nil {
+				if err == ErrNoUpdateAvailable {
+					// That's OK, let's make sure.
+					newAsset := testClient.latestAssetsMap[os][arch]
+					if asset != newAsset {
+						t.Fatal("CheckForUpdate said no update was available!")
+					}
+				} else {
+					t.Fatal("CheckForUpdate: ", err)
+				}
+			}
+
+			if r.PatchType != PATCHTYPE_NONE {
+				t.Fatal("Expecting no patch.")
+			}
+
+			if r.Version != testClient.latestAssetsMap[os][arch].v.String() {
+				t.Fatal("Expecting %v, got %v.", testClient.latestAssetsMap[os][arch].v, r.Version)
+			}
+		}
+	}
+}
+
+func TestDownloadManotoBetaAndUpgradeIt(t *testing.T) {
```

Comment:
This is going to simulate and test the case when a `2.0.0-beta8+manoto` client asks the server for update instructions.